velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -114,13 +114,38 @@ VeloxToSubstraitExprConvertor::toSubstraitExpr(
   ::substrait::Expression_FieldReference* substraitFieldExpr =
       google::protobuf::Arena::CreateMessage<
           ::substrait::Expression_FieldReference>(&arena);
-
-  std::string exprName = fieldExpr->name();
-
   ::substrait::Expression_ReferenceSegment_StructField* directStruct =
       substraitFieldExpr->mutable_direct_reference()->mutable_struct_field();
+  std::string exprName = fieldExpr->name();
+  // FieldAccessTypedExpr represents one of two things: a leaf in an expression
+  // or a dereference expression(fieldExpr->isInputColumn() == false)
+  // for a leaf in an expression, find idx from child by exprName.
+  // for a dereference expression, find idx from every child by exprName.
+  if (fieldExpr->isInputColumn()) {
+    auto idx = inputType->getChildIdxIfExists(exprName);
+    if (idx.has_value()) {
```

Comment:
What happens if this is false? Should we throw an exception?

Reply:
Yes, I think we should throw an exception.